components/org.wso2.analytics.is/src/main/java/org/wso2/analytics/is/siddhi/extension/JsonFlatMapEventDuplicatorStreamProcessor.java

```diff
@@ -0,0 +1,131 @@
+            if (attributeExpressionExecutors[0] instanceof VariableExpressionExecutor) {
+                if (attributeExpressionExecutors[0].getReturnType() == Attribute.Type.STRING) {
+                    expressionExecutor = (VariableExpressionExecutor) attributeExpressionExecutors[0];
+                } else {
+                    throw new ExecutionPlanValidationException("Json Flat Map Event Duplicator stream Processor 1st " +
+                            "attribute should be String, but found " + attributeExpressionExecutors[0].getReturnType());
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Event duplicate stream Processor 1st parameter " +
+                        "needs to be dynamic variable but found a constant attribute " +
+                        attributeExpressionExecutors[0].getClass().getCanonicalName());
+            }
+        } else {
+            throw new ExecutionPlanValidationException("Json Flat Map Event Duplicator stream Processor should only " +
+                    "have one parameter (<string> jsonString) but found "
+                    + attributeExpressionExecutors.length + " input attributes");
+        }
+
+        List<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(new Attribute("key", Attribute.Type.STRING));
+        attributeList.add(new Attribute("value", Attribute.Type.STRING));
+        return attributeList;
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+
+        ComplexEventChunk<StreamEvent> newStreamEventChunk = new ComplexEventChunk<>(false);
+        synchronized (this) {
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                String jsonString = (String) expressionExecutor.execute(streamEvent);
+                if (StringUtils.isNotBlank(jsonString)) {
+                    Map<String, String> map =
+                            new Gson().fromJson(jsonString, new TypeToken<Map<String, String>>() {
+                            }.getType());
+
+                    for (Map.Entry<String, String> entry : map.entrySet()) {
+                        StreamEvent clonedEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                        complexEventPopulater
+                                .populateComplexEvent(clonedEvent, new Object[]{entry.getKey(), entry.getValue()});
+                        newStreamEventChunk.add(clonedEvent);
+                    }
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Json Flat Map Event Duplicator stream processor " +
+                                "ignored request to duplicate null object");
```

Comment:
I really can't understand what this means :)
Ideally, what this does is cloning a new event with already read it seems.
If so, in the debug log say 'Ignored cloning a new event for empty event received' or something similar

Reply:
This extension doesn't duplicate events. This is similar to the event duplicator created for creating multiple events for a roles list separated by commas. The difference is that the string, in this case, is a JSON map of key-value pairs. This extension (As mentioned in the PR description) creates multiple events (one for each key-value pair). We do not pass an event into the extension to be duplicated. It's the key-value pairs that we duplicate and return as events.

Therefore, considering your suggestion as well I changed the message to "Ignored creating events from empty JSON string"